### PR TITLE
mcstaging/imc2mc

### PR DESCRIPTION
**Pull request by MargotCh**
## PR checklist

Solves this issue: https://github.com/nf-core/modules/issues/6142 

- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR. Test data was added with PR https://github.com/nf-core/test-datasets/pull/1280
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

Comment: Did not add conda, works with Docker and Singularity which is sufficient for our workflow, as we won't run it with conda.

### CODE REVIEW

File: modules/nf-core/mcstaging/imc2mc/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    file(params.modules_testdata_base_path + 'imaging/staging/imaging_mass_cytometry/acquisition.txt', checkIfExists: true),
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.tif },
```
Comment (kbestak) [inline modules/nf-core/mcstaging/imc2mc/tests/main.nf.test:28]:
```suggestion
                { assert snapshot(file(process.out.tif[0][1]).name,
                    process.out.versions).match() }
```
And update snapshot

File: modules/nf-core/mcstaging/imc2mc/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                { assert snapshot(process.out.versions).match() }
```
Comment (kbestak) [inline modules/nf-core/mcstaging/imc2mc/tests/main.nf.test:29]:
```suggestion
```

File: modules/nf-core/mcstaging/imc2mc/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                    file(params.modules_testdata_base_path + 'imaging/staging/imaging_mass_cytometry/acquisition.txt', checkIfExists: true),
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.tif },
```
Comment (kbestak) [inline modules/nf-core/mcstaging/imc2mc/tests/main.nf.test:53]:
```suggestion
                { assert snapshot(file(process.out.tif[0][1]).name,
                    process.out.versions).match() }
```

File: modules/nf-core/mcstaging/imc2mc/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                { assert snapshot(process.out.versions).match() }
```
Comment (kbestak) [inline modules/nf-core/mcstaging/imc2mc/tests/main.nf.test:54]:
```suggestion
```

**Review by kbestak — COMMENTED** [review]
Looks good overall, I left small formatting corrections and more specific snapshot definition

**Comment by MargotCh** [comment]
@kbestak I added the recommended changes, thanks for your comment!

**Review by kbestak — APPROVED** [review]
Thanks @MargotCh, LGTM!

**Pull request merged**